File: app/src/main/java/com/stupidtree/hichat/data/AppDatabase.java
package com.stupidtree.hichat.data;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.stupidtree.hichat.data.model.ChatMessage;
import com.stupidtree.hichat.data.model.Conversation;
import com.stupidtree.hichat.data.source.ChatMessageDao;
import com.stupidtree.hichat.data.source.ConversationDao;

@Database(entities = {ChatMessage.class, Conversation.class}, version = 1)
@androidx.room.TypeConverters({TypeConverters.class})
public abstract class AppDatabase extends RoomDatabase {
    public abstract ChatMessageDao chatMessageDao();
    public abstract ConversationDao conversationDao();
    private static volatile AppDatabase INSTANCE;

    public static AppDatabase getDatabase(Context context){
        if (INSTANCE == null) {
            synchronized (AppDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            AppDatabase.class, "cloudliter").build();
                }
            }
        }
        return INSTANCE;
    }

}

File: app/src/main/java/com/stupidtree/cloudliter/data/AppDatabase.kt
package com.stupidtree.cloudliter.data

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.stupidtree.cloudliter.data.model.*
import com.stupidtree.cloudliter.data.source.dao.*

@Database(entities = [ChatMessage::class, Conversation::class, UserRelation::class, UserProfile::class, ImageEntity::class,FaceResult::class], version = 1)
@androidx.room.TypeConverters(TypeConverters::class)
abstract class AppDatabase : RoomDatabase() {
    abstract fun chatMessageDao(): ChatMessageDao
    abstract fun conversationDao(): ConversationDao
    abstract fun userRelationDao():UserRelationDao
    abstract fun userProfileDao(): UserProfileDao
    abstract fun faceResultDao(): FaceResultDao
    abstract fun imageDao(): ImageDao
    companion object {
        @Volatile
        private var INSTANCE: AppDatabase? = null
        @JvmStatic
        fun getDatabase(context: Context): AppDatabase {
            if (INSTANCE == null) {
                synchronized(AppDatabase::class.java) {
                    if (INSTANCE == null) {
                        INSTANCE = Room.databaseBuilder(context.applicationContext,
                                AppDatabase::class.java, "cloudliter").build()
                    }
                }
            }
            return INSTANCE!!
        }
    }
}